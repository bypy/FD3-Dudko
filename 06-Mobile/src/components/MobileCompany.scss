@use "sass:math";

$bg-table: white;
$bg-header: #e9e9ed;
$header-cells-count: 7;
$header-text-color: black;
$header-font-size: 1rem;
$table-border-color: darkslategray;
$table-outer-border-width: 2px;
$table-inner-border-width: 1px;
$border-inner: solid $table-inner-border-width $table-border-color;
$border-outer: solid $table-outer-border-width $table-border-color;


.MobileCompany {

  box-sizing: border-box;
  *, *:before, *:after {
    box-sizing: inherit;
  }

  background: $bg-table;
  border: $border-outer;


  .caption {
    height: 3em;
    background-color: $bg-header;
    border-bottom: $border-outer;
    text-transform: uppercase;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .headerWrapper {
    position: sticky;
    top: 0;
    z-index: 20;
    background-color: $bg-header;
    border-bottom: $border-outer;
  }

  .mCompanyHeader {
    display: flex;
    flex-flow: row wrap;
    align-items: center;
    word-wrap: normal;
    line-height: 100%;

    .cell {
      width: calc(100% / #{$header-cells-count});
      padding: 0.5em 0.5em;
      border-right: $border-inner;
      color: $header-text-color;
      text-align: center;
      word-wrap: break-word;
      word-break: break-all;
      overflow-x: hidden;
      font-size: $header-font-size;

      &:last-child {
        border-right: 0;
      }
    }

  }

  .addRecordBtn {
    background: $bg-header;
    border-top: $border-inner;
    padding: 1em 0;
    text-align: center;

    button {
      padding: 5px 10px;
      font-size: inherit;
      font-family: inherit;
    }
  }

}

@media all and (max-width: 767px) {
  .MobileCompany {

    .caption {
      font-size: math.div($header-font-size * 3, 4);
    }

    .mCompanyHeader {
      min-height: 70px;
      align-items: normal;

      .cell {
        width: calc(100% / #{$header-cells-count - 1});
        display: flex;
        flex-flow: column nowrap;
        justify-content: center;
        padding: 0.25em 0.25em;
        font-size: math.div($header-font-size * 3, 4);

        &.first {
          width: 100%;
          border-right: 0;
          border-bottom: solid $table-inner-border-width $table-border-color;
        }
      }
    }
  }
}

@media all and (max-width: 430px) {

  .MobileCompany {

    .mCompanyHeader {
      counter-reset: property-number;
      min-height: 140px;

      .cell {
        display: flex;
        flex-flow: row nowrap;
        width: 100%;
        border-right: 0;
        counter-increment: property-number;

        &::before {
          content: counter(property-number)". ";
          margin-right: 2ex;
          color: dimgray;
          font-size: 0.75em;
          transform: translateY(2px);
        }
        &.first {
          border-bottom: 0;
        }
      }

    }

  }

}

