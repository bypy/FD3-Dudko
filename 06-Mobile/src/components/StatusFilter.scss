$button-accent: rgb(200, 10, 10);

.StatusFilter {
  box-sizing: border-box;

  *, *:before, *:after {
    box-sizing: inherit;
  }

  .buttons-tab {
    max-width: 30em;
    display: flex;
    justify-content: left;
  }

  button {
    font-size: 1em;
    min-width: 10em;
    padding: .5em .25em;

    &.active {
      color: $button-accent;
    }
  }

  button:nth-last-child(n+2) {
    margin-right: 1em;
  }


}