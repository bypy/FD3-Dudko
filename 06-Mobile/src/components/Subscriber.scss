@use "sass:math";

$cells-total-count: 7;
$bg-table: white;
$item-bg-color: #e9e9ed;
$item-text-color: black;
$item-font-size: 1rem;
$selected-item-bg-color: #ee8cfa;
$item-border-width: 1px;
$item-border-double-width: 2px;
$item-border-color: darkslategray;
$item-border-style: solid;
$border: $item-border-width $item-border-style $item-border-color;
$border-bold: $item-border-double-width $item-border-style $item-border-color;

.Subscriber {
  font-size: 100%;
  display: flex;
  flex-flow: row wrap;
  transition: background-color 0.1s ease-in-out;

  &:nth-child(odd) {
    background: $item-bg-color;

    .cell.first {
      z-index: 11;
      background: $item-bg-color;
    }
  }

  &.selected {
    background-color: $selected-item-bg-color;

    .cell.first {
      background-color: $selected-item-bg-color;
    }
  }

  .cell {
    width: calc(100% / #{$cells-total-count});
    padding: 0.5em 0.5em;
    border-right: $border;
    color: $item-text-color;

    text-align: center; // horizontal align
    word-wrap: break-word;
    word-break: break-all;
    overflow-x: hidden;
    display: flex;
    flex-direction: column;
    justify-content: center; // vertical align

    &.left-align {
      text-align: left;
    }

    &:last-child {
      border-right: 0;
    }

    &.first {
      background: $bg-table;
      z-index: 11;
    }

    &.blocked {
      background-color: crimson;
      color: white;
    }

    .actionBtn {
      display: flex;
      align-self: center;
      padding: 0 1ex;
      font-family: inherit;
    }

    &.column {
      align-items: center;
    }


  }
}

@media all and (max-width: 767px) {
  .Subscriber {

    &:not(:last-child) {
      border-bottom: $border;
    }

    .cell {
      width: calc(100% / #{$cells-total-count - 1});
      font-size: math.div($item-font-size * 3, 4);

      &.first {
        position: sticky;
        z-index: 10;
        top: 69px;
        width: 100%;
        border-right: 0;
        border-bottom: $border;
        filter: opacity(0.96);
        text-align: center;
      }
    }

  }
}


@media all and (max-width: 430px) {
  .Subscriber {

    counter-reset: property-number;
    position: relative;

    .cell {
      width: 100%;
      font-size: $item-font-size;
      border-right: 0;
      counter-increment: property-number;
      padding-left: 3ex;

      &::before {
        content: counter(property-number) ": ";
        position: absolute;
        left: 0.25em;
        color: dimgray;
        font-size: 0.75em;
      }

      &.first {
        top: 140px;
      }

      &.column {
        flex-direction: row;
        margin-bottom: 0.5em;
      }

      .actionBtn {
        &:last-child {
          margin: 0 0 0 0.5em;
        }
      }
    }
  }
}
