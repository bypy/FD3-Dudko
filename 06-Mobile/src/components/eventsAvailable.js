const LIFECYCLE_EVENT = 'life_cycle_event';
const RENDER_EVENT = 'render_event';
const FILTER_CHANGE = 'status_filter_changed_event';
const ITEM_EDIT = 'subscriber_edit_in_progress';
const ITEM_EDIT_COMPLETE = 'subscriber_clicked_event';
const ITEM_DELETE = 'subscriber_delete_event';
const ITEM_SAVE = 'subscriber_save_event';
const SET_FORM_INVALID = 'some_user_inputs_are_invalid';


export {
  LIFECYCLE_EVENT,
  RENDER_EVENT,
  FILTER_CHANGE,
  ITEM_EDIT_COMPLETE,
  ITEM_EDIT,
  ITEM_DELETE,
  ITEM_SAVE,
  SET_FORM_INVALID
};