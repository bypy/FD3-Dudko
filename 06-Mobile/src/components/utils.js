import { LOG_MODE } from './logModes';

function concatAndHash() {
  return Math.abs(
    Array.from(arguments)
      .join('')
      .split('')
      .reduce((a, b) => ((a << 5) - a + b.charCodeAt(0)) | 0, 0)
  );
}

function makeLogger(output, mode) {
  if (!output || !output.log) throw Error('Provide logging entity with .info and .log methods implementation!');
  return mode === LOG_MODE.INFO
    ? (message) => output.info(message)
    : (message) => output.log(`${LOG_MODE.DEBUG}: ${message}`);
}

function subscriberFilterFunc(currentFilter, subscriber) {
  return (
    currentFilter === 0 ||
    (currentFilter === 1 && !(subscriber.balance < 0)) ||
    (currentFilter === 2 && subscriber.balance < 0)
  );
}

const newSubscriberTempId = 0;
const newSubscriberInitBalance = 0;

export { concatAndHash, makeLogger, subscriberFilterFunc, newSubscriberTempId, newSubscriberInitBalance };
