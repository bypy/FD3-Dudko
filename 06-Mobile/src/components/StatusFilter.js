import React from 'react';
import { PropTypes } from 'prop-types';

import { eventBus } from './eventBus';
import { LIFECYCLE_EVENT, RENDER_EVENT, FILTER_CHANGE } from './eventsAvailable';

import './StatusFilter.scss';

export default class StatusFilter extends React.PureComponent {
  static propTypes = {
    currentFilter: PropTypes.number.isRequired,
    companyInEditMode: PropTypes.bool.isRequired,
    btn: PropTypes.shape({}),
  };

  static defaultProps = {
    btn: {
      all: 'Все',
      active: 'Активные',
      blocked: 'Заблокированные',
    },
  };

  filterChanged = (EO) => {
    EO.stopPropagation();
    EO.preventDefault();
    eventBus.emit(FILTER_CHANGE, EO.target.name);
  };

  render() {
    eventBus.emit(RENDER_EVENT, `${this.constructor.name} component RENDER`);
    return (
      <div className="StatusFilter">
        <span className="buttons-tab" onClick={this.filterChanged}>
          <button
            name="0"
            className={this.props.currentFilter === 0 ? 'active' : null}
            disabled={this.props.companyInEditMode}
          >
            {this.props.btn.all}
          </button>
          <button
            name="1"
            className={this.props.currentFilter === 1 ? 'active' : null}
            disabled={this.props.companyInEditMode}
          >
            {this.props.btn.active}
          </button>
          <button
            name="2"
            className={this.props.currentFilter === 2 ? 'active' : null}
            disabled={this.props.companyInEditMode}
          >
            {this.props.btn.blocked}
          </button>
        </span>
      </div>
    );
  }
}
