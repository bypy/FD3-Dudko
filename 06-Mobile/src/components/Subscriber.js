import React from 'react';
import { Fragment } from 'react';

import PropTypes from 'prop-types';

import { eventBus } from './eventBus';
import {
  LIFECYCLE_EVENT,
  RENDER_EVENT,
  ITEM_EDIT_COMPLETE,
  ITEM_EDIT,
  ITEM_DELETE,
  ITEM_SAVE,
  PREVENT_LOSING_CHANGES,
  SET_FORM_INVALID,
} from './eventsAvailable';
import { concatAndHash, newSubscriberTempId } from './utils';
import { VALIDATION_TYPES as v } from './validator';

import './Subscriber.scss';

export default class Subscriber extends React.PureComponent {
  static propTypes = {
    data: PropTypes.shape({
      id: PropTypes.number.isRequired,
      lastName: PropTypes.string.isRequired,
      firstName: PropTypes.string.isRequired,
      surName: PropTypes.string.isRequired,
      balance: PropTypes.number.isRequired,
    }), // is null in add subscriber mode
    companyInEditMode: PropTypes.bool,
    btn: PropTypes.shape({}),
  };

  static defaultProps = {
    companyInEditMode: null,
    btn: {
      edit: 'Редактировать',
      delete: 'Удалить',
    },
  };

  state = {
    subscriberInEditMode: false,
  }; // derivedProps: subscriberInEditMode, isBlockedCustomer

  static getDerivedStateFromProps(props, state) {
    eventBus.emit(LIFECYCLE_EVENT, `getDerivedStateFromProps from Subscriber id=${props.data.id} component`);
    let wasUnderEdit = state.subscriberInEditMode;
    let newEditInProgress = props.companyInEditMode;
    let dropEditMode = wasUnderEdit && newEditInProgress;
    return {
      subscriberInEditMode: dropEditMode, // other subscriber going to be changed
      isBlockedCustomer: props.data.balance < 0,
    };
  }

  componentDidMount() {
    eventBus.addListener(ITEM_SAVE, this.saveSubscriber);
  }

  componentWillUnmount() {
    eventBus.removeListener(ITEM_SAVE, this.saveSubscriber);
  }

  editSubscriber = (EO) => {
    EO.preventDefault();
    if (this.props.companyInEditMode) {
      if (confirm('Внесенные изменения будут потеряны! OK: сохранить')) {
        eventBus.emit(PREVENT_LOSING_CHANGES, true);
      } else {
        this.setState({ subscriberInEditMode: true });
        eventBus.emit(ITEM_EDIT);
        eventBus.emit(PREVENT_LOSING_CHANGES, false);
      }
    } else {
      this.setState({ subscriberInEditMode: true });
      eventBus.emit(ITEM_EDIT);
    }
  };

  unsetEditState = () => {
    this.setState({ subscriberInEditMode: false });
  };

  saveSubscriber = (argList) => {
    const [clearWarningsFlag, validationFunc] = argList;
    if (!this.state.subscriberInEditMode) return; // fires only for currently editing input field!
    clearWarningsFlag && this.hideErrors();
    const [errors, validUserData] = this.collectSubscriberErrorsAndData(validationFunc);
    if (errors.length) {
      this.showError(errors);
      eventBus.emit(SET_FORM_INVALID);
    } else {
      eventBus.emit(ITEM_EDIT_COMPLETE, validUserData);
    }
  };

  showError = (errorList) => {
    errorList.forEach((e) => {
      const { ref, errorText } = e;
      const errNode = document.createElement('span');
      errNode.className = 'invalid';
      errNode.textContent = errorText;
      ref.parentNode.appendChild(errNode);
    });
  };

  hideErrors = () => {
    const inputRefs = [this.lastNameRef, this.firstNameRef, this.surNameRef, this.balanceRef].filter((r) => !!r);
    inputRefs.forEach((ref) => {
      const errNode = ref.parentNode.querySelector('.invalid');
      errNode && ref.parentNode.removeChild(errNode);
    });
  };

  deleteRecord = () => {
    if (confirm(`Удаляем абонента ${this.props.data.firstName} c #${this.props.data.id}?`)) {
      eventBus.emit(ITEM_DELETE, this.props.data.id);
    }
  };

  collectSubscriberErrorsAndData = (getValidationError) => {
    let errorList = [];
    let lastName, lastNameError;
    if (this.lastNameRef) {
      lastNameError = getValidationError(this.lastNameRef.value, [v.notEmpty, v.isString, v.isWord, v.minLength(2)]);
      lastNameError && errorList.push({ ref: this.lastNameRef, errorText: lastNameError });
      lastName = this.lastNameRef.value;
    }
    let firstName, firstNameError;
    if (this.firstNameRef) {
      firstNameError = getValidationError(this.firstNameRef.value, [v.notEmpty, v.isString, v.isWord]);
      firstNameError && errorList.push({ ref: this.firstNameRef, errorText: firstNameError });
      firstName = this.firstNameRef.value;
    }
    let surName, surNameError;
    if (this.firstNameRef) {
      surNameError = getValidationError(this.surNameRef.value, [v.isWord]);
      surNameError && errorList.push({ ref: this.surNameRef, errorText: surNameError });
      surName = this.surNameRef.value;
    }
    let balance, balanceError;
    if (this.balanceRef) {
      balanceError = getValidationError(this.balanceRef.value, [v.notEmpty, v.isNumber]);
      balanceError && errorList.push({ ref: this.balanceRef, errorText: balanceError });
      balance = balanceError ? this.balanceRef.value : Number(this.balanceRef.value);
    }
    return [errorList, { ...this.props.data, lastName, firstName, surName, balance }];
  };

  setLastNameRef = (ref) => {
    this.lastNameRef = ref;
  };
  setFirstNameRef = (ref) => {
    this.firstNameRef = ref;
  };
  setSurNameRef = (ref) => {
    this.surNameRef = ref;
  };
  setBalanceRef = (ref) => {
    this.balanceRef = ref;
  };

  getStatusNameByCode = (isBlockedCustomer) => (isBlockedCustomer ? 'blocked' : 'active');

  render() {
    eventBus.emit(RENDER_EVENT, `${this.constructor.name} id=${this.props.data.id} component RENDER`);
    return (
      <div className="Subscriber" role="row">
        <div className="cell first left-align" role="cell">
          {this.state.subscriberInEditMode && (
            <Fragment>
              <input
                className="userDataInput"
                size={this.props.data.lastName.toString().length || 8} // 3 - default input width for empty content
                name="lastName"
                defaultValue={this.props.data.lastName}
                ref={this.setLastNameRef}
              />
              <span className="customerId">ID: {this.props.data.id}</span>
            </Fragment>
          )}
          {!this.state.subscriberInEditMode && (
            <Fragment>
              <span>{this.props.data.lastName}</span>
              <span className="customerId">ID: {this.props.data.id}</span>
            </Fragment>
          )}
        </div>
        <div className="cell left-align" role="cell">
          {this.state.subscriberInEditMode && (
            <input
              className="userDataInput"
              size={this.props.data.firstName.toString().length || 8} // 3 - default input width for empty content
              defaultValue={this.props.data.firstName}
              ref={this.setFirstNameRef}
            />
          )}
          {!this.state.subscriberInEditMode && this.props.data.firstName}
        </div>
        <div className="cell left-align" role="cell">
          {this.state.subscriberInEditMode && (
            <input
              className="userDataInput"
              size={this.props.data.surName.toString().length || 8} // 3 - default input width for empty content
              defaultValue={this.props.data.surName}
              ref={this.setSurNameRef}
            />
          )}
          {!this.state.subscriberInEditMode && this.props.data.surName}
        </div>
        <div className="cell" role="cell">
          {this.state.subscriberInEditMode && (
            <input
              className="userDataInput"
              size={this.props.data.balance.toString().length || 8} // 3 - default input width for empty content
              defaultValue={this.props.data.balance}
              ref={this.setBalanceRef}
            />
          )}
          {!this.state.subscriberInEditMode && this.props.data.balance}
        </div>
        <div className={this.state.isBlockedCustomer ? 'cell blocked' : 'cell'} role="cell">
          {this.getStatusNameByCode(this.state.isBlockedCustomer)}
        </div>
        <div className="cell" role="cell">
          <button
            className="actionBtn"
            onClick={this.editSubscriber}
            disabled={this.state.subscriberInEditMode || this.props.companyInEditMode}
          >
            {this.props.btn.edit}
          </button>
        </div>
        <div className="cell" role="cell">
          <button className="actionBtn" onClick={this.deleteRecord}>
            {this.props.btn.delete}
          </button>
        </div>
      </div>
    );
  }
}
