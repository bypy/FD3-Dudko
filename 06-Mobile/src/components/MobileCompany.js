import React from 'react';
import { Fragment } from 'react';
import { PropTypes } from 'prop-types';

import Subscriber from './Subscriber';

import './IShop.scss';
import './IShopItem.scss';

export default class MobileCompany extends React.Component {
  static propTypes = {
    subscribers: PropTypes.arrayOf(
      PropTypes.shape({
        lastName: PropTypes.string.isRequired,
        firstName: PropTypes.string.isRequired,
        surName: PropTypes.string.isRequired,
        balance: PropTypes.number.isRequired,
        status: PropTypes.string.isRequired,
      }).isRequired
    ).isRequired,
    loggerMode: PropTypes.string,
    classname: PropTypes.string,
  };

  static defaultProps = {
    btn: {
      company1: 'A1',
      company2: 'MTS',
      all: 'Все',
      active: 'Активные',
      blocked: 'Заблокированные',
      edit: 'Редактировать',
      delete: 'Удалить',
      add: 'Добавить клиента',
    },
    // header: ['Фамилия', 'Имя', 'Отчество', 'Баланс', 'Статус', 'Редактировать', 'Удалить'],
    header: Array.prototype.map.call(
      ['Фамилия', 'Имя', 'Отчество', 'Баланс', 'Статус', 'Редактировать', 'Удалить'],
      (property, index) => (
        <div key={index} className={index === 0 ? 'cell first' : 'cell'} role="columnheader">
          {property}
        </div>
      )
    ),
    loggerMode: 'info',
    classname: 'IShop',
  };

  state = {
    subscribers: this.props.subscribers.map((subscriber) => <Subscriber key={subscriber.id} data={subscriber} />),
  };

  componentWillMount() {

  }

  render() {
    return (
      <Fragment>
        <div className={this.props.classname} role="table">
          <div className="headerWrapper">
            <div className="iShopHeader" role="row">
              {this.props.header}
            </div>
          </div>
          {this.state.subscribers}
        </div>
      </Fragment>
    );
  }
}
