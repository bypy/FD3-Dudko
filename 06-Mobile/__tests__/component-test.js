import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import MobileCompany from '../src/components/MobileCompany';
import Subscriber from '../src/components/Subscriber';
import StatusFilter from '../src/components/StatusFilter';
import { concatAndHash, subscriberFilterFunc } from '../src/components/utils';

const subscribers = require('../src/data/subscribers.json').map((subscriber) => {
  subscriber.id = concatAndHash(subscriber.lastName, subscriber.firstName, subscriber.surName);
  return subscriber;
});

// mock confirm action
global.confirm = jest.fn(() => true);

describe('удаление абонента из MobileCompany', () => {
  let component = renderer.create(<MobileCompany company={'A1'} subscribers={subscribers} />);
  let componentInstance = component.root;

  test('рендеринг перед удалением абонента', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  let deleteButtons = componentInstance.findAll((el) => el.type === 'button' && el.props.name === 'delete');
  let subscriberIndexToDelete = 2; // remove third subscriber from list
  let deleteButtonElem = deleteButtons[subscriberIndexToDelete];
  deleteButtonElem.props.onClick(undefined);

  test('рендеринг после удаления абонента', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  let subscribersList = componentInstance.findAllByType(Subscriber);
  test('длина списка абонентов уменьшилась на один', () => {
    expect(subscribersList.length).toEqual(componentInstance.props.subscribers.length - 1);
  });

  test('выбранный для удаления id абонента отсутствует среди отображаемых абонентов', () => {
    let deletedId = null;
    componentInstance.props.subscribers.forEach((subscriber, index) => {
      if (index === subscriberIndexToDelete) deletedId = subscriber.id;
    });
    expect(subscribersList.filter((el) => el.props.data.id === deletedId)).toEqual([]);
  });

  component.unmount();
});


describe('фильтрация абонентов MobileCompany', () => {
  let component = renderer.create(<MobileCompany company={'A1'} subscribers={subscribers} />);
  let componentInstance = component.root;
  let subscribersList,
    filteredList;

  test('рендеринг без фильтра', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  let wrapper = mount(<MobileCompany company={'A1'} subscribers={subscribers} />);
  let activeFilterCode = 1;
  wrapper.find(StatusFilter).find('.buttons-tab').simulate('click', {
    target: {
      name: activeFilterCode
    }
  });
  subscribersList = componentInstance.findAllByType(Subscriber);
  filteredList = componentInstance.props.subscribers.filter(s => subscriberFilterFunc(activeFilterCode, s));
  test('длина списка активных абонентов', () => {
    expect(subscribersList.length).toEqual(filteredList.length);
  });
  test('рендеринг списка активных абонентов', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
