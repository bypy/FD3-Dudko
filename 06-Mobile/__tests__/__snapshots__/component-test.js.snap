// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`удаление абонента из MobileCompany рендеринг перед удалением абонента 1`] = `null`;

exports[`удаление абонента из MobileCompany рендеринг после удаления абонента 1`] = `null`;

exports[`фильтрация абонентов MobileCompany рендеринг без фильтра 1`] = `
Array [
  <span>
    Компания: A1
  </span>,
  <hr />,
  <div
    className="StatusFilter"
  >
    <span
      className="buttons-tab"
      onClick={[Function]}
    >
      <button
        className={null}
        disabled={false}
        name="0"
      >
        Все
      </button>
      <button
        className="active"
        disabled={false}
        name="1"
      >
        Активные
      </button>
      <button
        className={null}
        disabled={false}
        name="2"
      >
        Заблокированные
      </button>
    </span>
  </div>,
  <hr />,
  <div
    className="MobileCompany"
    role="table"
  >
    <div
      className="headerWrapper"
    >
      <div
        className="mCompanyHeader"
        role="row"
      >
        <div
          className="cell first"
        >
          Фамилия*
        </div>
        <div
          className="cell"
        >
          Имя*
        </div>
        <div
          className="cell"
        >
          Отчество
        </div>
        <div
          className="cell"
        >
          Баланс*
        </div>
        <div
          className="cell"
        >
          Статус
        </div>
        <div
          className="cell"
        >
          Редактировать
        </div>
        <div
          className="cell"
        >
          Удалить
        </div>
      </div>
    </div>
    <div
      className="Subscriber"
      role="row"
    >
      <div
        className="cell first left-align"
        role="cell"
      >
        <span>
          Иванов
        </span>
        <span
          className="customerId"
        >
          ID: 
          1730812076
        </span>
      </div>
      <div
        className="cell left-align"
        role="cell"
      >
        Иван
      </div>
      <div
        className="cell left-align"
        role="cell"
      >
        Иванович
      </div>
      <div
        className="cell"
        role="cell"
      >
        200
      </div>
      <div
        className="cell active"
        role="cell"
      >
        active
      </div>
      <div
        className="cell"
        role="cell"
      >
        <button
          className="actionBtn"
          disabled={false}
          name="edit"
          onClick={[Function]}
        >
          Редактировать
        </button>
      </div>
      <div
        className="cell"
        role="cell"
      >
        <button
          className="actionBtn"
          disabled={false}
          name="delete"
          onClick={[Function]}
        >
          Удалить
        </button>
      </div>
    </div>
    <div
      className="Subscriber"
      role="row"
    >
      <div
        className="cell first left-align"
        role="cell"
      >
        <span>
          Сидоров
        </span>
        <span
          className="customerId"
        >
          ID: 
          2085433142
        </span>
      </div>
      <div
        className="cell left-align"
        role="cell"
      >
        Сидор
      </div>
      <div
        className="cell left-align"
        role="cell"
      >
        Сидорович
      </div>
      <div
        className="cell"
        role="cell"
      >
        250
      </div>
      <div
        className="cell active"
        role="cell"
      >
        active
      </div>
      <div
        className="cell"
        role="cell"
      >
        <button
          className="actionBtn"
          disabled={false}
          name="edit"
          onClick={[Function]}
        >
          Редактировать
        </button>
      </div>
      <div
        className="cell"
        role="cell"
      >
        <button
          className="actionBtn"
          disabled={false}
          name="delete"
          onClick={[Function]}
        >
          Удалить
        </button>
      </div>
    </div>
    <div
      className="Subscriber"
      role="row"
    >
      <div
        className="cell first left-align"
        role="cell"
      >
        <span>
          Петров
        </span>
        <span
          className="customerId"
        >
          ID: 
          372639379
        </span>
      </div>
      <div
        className="cell left-align"
        role="cell"
      >
        Петр
      </div>
      <div
        className="cell left-align"
        role="cell"
      >
        Петрович
      </div>
      <div
        className="cell"
        role="cell"
      >
        180
      </div>
      <div
        className="cell active"
        role="cell"
      >
        active
      </div>
      <div
        className="cell"
        role="cell"
      >
        <button
          className="actionBtn"
          disabled={false}
          name="edit"
          onClick={[Function]}
        >
          Редактировать
        </button>
      </div>
      <div
        className="cell"
        role="cell"
      >
        <button
          className="actionBtn"
          disabled={false}
          name="delete"
          onClick={[Function]}
        >
          Удалить
        </button>
      </div>
    </div>
  </div>,
  <hr />,
  <div
    className="ctrlButtons"
  >
    <span
      className="buttons-tab"
    >
      <button
        id="add"
        onClick={[Function]}
      >
        Добавить клиента
      </button>
      <button
        disabled={true}
        id="save"
        onClick={[Function]}
      >
        Сохранить клиента
      </button>
      <button
        disabled={true}
        id="cancel"
        onClick={[Function]}
      >
        Отмена
      </button>
    </span>
  </div>,
]
`;

exports[`фильтрация абонентов MobileCompany рендеринг списка активных абонентов 1`] = `
Array [
  <span>
    Компания: A1
  </span>,
  <hr />,
  <div
    className="StatusFilter"
  >
    <span
      className="buttons-tab"
      onClick={[Function]}
    >
      <button
        className={null}
        disabled={false}
        name="0"
      >
        Все
      </button>
      <button
        className="active"
        disabled={false}
        name="1"
      >
        Активные
      </button>
      <button
        className={null}
        disabled={false}
        name="2"
      >
        Заблокированные
      </button>
    </span>
  </div>,
  <hr />,
  <div
    className="MobileCompany"
    role="table"
  >
    <div
      className="headerWrapper"
    >
      <div
        className="mCompanyHeader"
        role="row"
      >
        <div
          className="cell first"
        >
          Фамилия*
        </div>
        <div
          className="cell"
        >
          Имя*
        </div>
        <div
          className="cell"
        >
          Отчество
        </div>
        <div
          className="cell"
        >
          Баланс*
        </div>
        <div
          className="cell"
        >
          Статус
        </div>
        <div
          className="cell"
        >
          Редактировать
        </div>
        <div
          className="cell"
        >
          Удалить
        </div>
      </div>
    </div>
    <div
      className="Subscriber"
      role="row"
    >
      <div
        className="cell first left-align"
        role="cell"
      >
        <span>
          Иванов
        </span>
        <span
          className="customerId"
        >
          ID: 
          1730812076
        </span>
      </div>
      <div
        className="cell left-align"
        role="cell"
      >
        Иван
      </div>
      <div
        className="cell left-align"
        role="cell"
      >
        Иванович
      </div>
      <div
        className="cell"
        role="cell"
      >
        200
      </div>
      <div
        className="cell active"
        role="cell"
      >
        active
      </div>
      <div
        className="cell"
        role="cell"
      >
        <button
          className="actionBtn"
          disabled={false}
          name="edit"
          onClick={[Function]}
        >
          Редактировать
        </button>
      </div>
      <div
        className="cell"
        role="cell"
      >
        <button
          className="actionBtn"
          disabled={false}
          name="delete"
          onClick={[Function]}
        >
          Удалить
        </button>
      </div>
    </div>
    <div
      className="Subscriber"
      role="row"
    >
      <div
        className="cell first left-align"
        role="cell"
      >
        <span>
          Сидоров
        </span>
        <span
          className="customerId"
        >
          ID: 
          2085433142
        </span>
      </div>
      <div
        className="cell left-align"
        role="cell"
      >
        Сидор
      </div>
      <div
        className="cell left-align"
        role="cell"
      >
        Сидорович
      </div>
      <div
        className="cell"
        role="cell"
      >
        250
      </div>
      <div
        className="cell active"
        role="cell"
      >
        active
      </div>
      <div
        className="cell"
        role="cell"
      >
        <button
          className="actionBtn"
          disabled={false}
          name="edit"
          onClick={[Function]}
        >
          Редактировать
        </button>
      </div>
      <div
        className="cell"
        role="cell"
      >
        <button
          className="actionBtn"
          disabled={false}
          name="delete"
          onClick={[Function]}
        >
          Удалить
        </button>
      </div>
    </div>
    <div
      className="Subscriber"
      role="row"
    >
      <div
        className="cell first left-align"
        role="cell"
      >
        <span>
          Петров
        </span>
        <span
          className="customerId"
        >
          ID: 
          372639379
        </span>
      </div>
      <div
        className="cell left-align"
        role="cell"
      >
        Петр
      </div>
      <div
        className="cell left-align"
        role="cell"
      >
        Петрович
      </div>
      <div
        className="cell"
        role="cell"
      >
        180
      </div>
      <div
        className="cell active"
        role="cell"
      >
        active
      </div>
      <div
        className="cell"
        role="cell"
      >
        <button
          className="actionBtn"
          disabled={false}
          name="edit"
          onClick={[Function]}
        >
          Редактировать
        </button>
      </div>
      <div
        className="cell"
        role="cell"
      >
        <button
          className="actionBtn"
          disabled={false}
          name="delete"
          onClick={[Function]}
        >
          Удалить
        </button>
      </div>
    </div>
  </div>,
  <hr />,
  <div
    className="ctrlButtons"
  >
    <span
      className="buttons-tab"
    >
      <button
        id="add"
        onClick={[Function]}
      >
        Добавить клиента
      </button>
      <button
        disabled={true}
        id="save"
        onClick={[Function]}
      >
        Сохранить клиента
      </button>
      <button
        disabled={true}
        id="cancel"
        onClick={[Function]}
      >
        Отмена
      </button>
    </span>
  </div>,
]
`;
